src/scene/shader-lib/chunks-wgsl/common/frag/fog.js

```diff
@@ -0,0 +1,44 @@
+export default /* glsl */`
+
+#if (FOG != NONE)
+    uniform fog_color : vec3f;
+    
+    #if (FOG == LINEAR)
+        uniform fog_start : f32;
+        uniform fog_end : f32;
+    #else
+        uniform fog_density : f32;
+    #endif
+#endif
+
+uniform dBlendModeFogFactor : f32;
+
+fn getFogFactor() -> f32 {
+    //let depth = gl_FragCoord.z / gl_FragCoord.w;
+    let depth = 1.0;
+
+
+
+
```

Comment:
Wow, you love new lines! 🤣

Reply:
hah, those are to make it obvious the code is unfinished .. a place I need to clean up before creating PR :)